Index: libs/libksane/libksane/ksane_viewer.cpp
===================================================================
--- libs/libksane/libksane/ksane_viewer.cpp	(.../tags/KDE/4.3.0/kdegraphics)	(wersja 1013435)
+++ libs/libksane/libksane/ksane_viewer.cpp	(.../branches/KDE/4.3/kdegraphics)	(wersja 1013435)
@@ -244,8 +244,8 @@
 {
     if ((d->selection->rect().width() >0.001) &&
         (d->selection->rect().height() > 0.001) &&
-        (d->pixmapItem->pixmap().width() - d->selection->rect().width() > 0.001) &&
-        (d->pixmapItem->pixmap().height() - d->selection->rect().height() > 0.001))
+        ((d->pixmapItem->pixmap().width() - d->selection->rect().width() > 0.001) ||
+        (d->pixmapItem->pixmap().height() - d->selection->rect().height() > 0.001)))
     {
         d->selection->setVisible(true);
     }
Index: gwenview/app/kipiinterface.cpp
===================================================================
--- gwenview/app/kipiinterface.cpp	(.../tags/KDE/4.3.0/kdegraphics)	(wersja 1013435)
+++ gwenview/app/kipiinterface.cpp	(.../branches/KDE/4.3/kdegraphics)	(wersja 1013435)
@@ -352,8 +352,7 @@
 
 
 void KIPIInterface::slotDirectoryChanged() {
-// TODO check if getting it from selected
-	emit currentAlbumChanged(d->mMainWindow->contextManager()->selection().count() >0);
+	emit currentAlbumChanged(true);
 }
 
 } //namespace
Index: gwenview/app/main.cpp
===================================================================
--- gwenview/app/main.cpp	(.../tags/KDE/4.3.0/kdegraphics)	(wersja 1013435)
+++ gwenview/app/main.cpp	(.../branches/KDE/4.3/kdegraphics)	(wersja 1013435)
@@ -39,7 +39,7 @@
 		"gwenview",        /* appname */
 		0,                 /* catalogName */
 		ki18n("Gwenview"), /* programName */
-		"2.3.0");          /* version */
+		"2.3.1");          /* version */
 	aboutData.setShortDescription(ki18n("An Image Viewer"));
 	aboutData.setLicense(KAboutData::License_GPL);
 	aboutData.setCopyrightStatement(ki18n("Copyright 2000-2009 Aurélien Gâteau"));
Index: gwenview/part/gvpart.cpp
===================================================================
--- gwenview/part/gvpart.cpp	(.../tags/KDE/4.3.0/kdegraphics)	(wersja 1013435)
+++ gwenview/part/gvpart.cpp	(.../branches/KDE/4.3/kdegraphics)	(wersja 1013435)
@@ -116,7 +116,7 @@
 		"gvpart",                /* appname */
 		"gwenview",              /* catalogName */
 		ki18n("Gwenview KPart"), /* programName */
-		"2.1");                  /* version */
+		"2.3.1");                /* version */
 	aboutData->setShortDescription(ki18n("An Image Viewer"));
 	aboutData->setLicense(KAboutData::License_GPL);
 	aboutData->setCopyrightStatement(ki18n("Copyright 2000-2008 Aurélien Gâteau"));
Index: gwenview/NEWS
===================================================================
--- gwenview/NEWS	(.../tags/KDE/4.3.0/kdegraphics)	(wersja 1013435)
+++ gwenview/NEWS	(.../branches/KDE/4.3/kdegraphics)	(wersja 1013435)
@@ -1,3 +1,7 @@
+# 2.3.1
+## Fixes
+- Ensure "Advanced Slideshow" plugin is always enabled.
+
 # 2.3.0
 ## New features
 - Gwenview can now play videos (bug #174522).
Index: thumbnailers/mobi/mobithumbnail.desktop
===================================================================
--- thumbnailers/mobi/mobithumbnail.desktop	(.../tags/KDE/4.3.0/kdegraphics)	(wersja 1013435)
+++ thumbnailers/mobi/mobithumbnail.desktop	(.../branches/KDE/4.3/kdegraphics)	(wersja 1013435)
@@ -4,7 +4,7 @@
 Name[ar]=مولد مصغرات Mobipocket
 Name[ca]=Miniaturitzador de Mobipocket
 Name[da]=Mobipocket-miniaturer
-Name[de]=M
+Name[de]=Mobipocket-Minibild
 Name[el]=Δημιουργία εικόνων επισκόπησης Mobipocket
 Name[es]=Generador de miniaturas Mobipocket
 Name[et]=Mobipocketi pisipildid
@@ -25,7 +25,7 @@
 Name[pa]=ਮੋਬਾਪਾਕਟ ਥੰਮਨੇਲਰ
 Name[pt]=Miniaturas para o Mobipocket
 Name[pt_BR]=Miniatura do Mobipocket
-Name[ru]=Создание эскизов Mobipocket
+Name[ru]=Создание миниатюр Mobipocket
 Name[sl]=Ustvarjalnik sličic za Mobipocket
 Name[sr]=Генератор сличица за Мобипокет
 Name[sr@latin]=Generator sličica za Mobipocket
Index: okular/core/document.cpp
===================================================================
--- okular/core/document.cpp	(.../tags/KDE/4.3.0/kdegraphics)	(wersja 1013435)
+++ okular/core/document.cpp	(.../branches/KDE/4.3/kdegraphics)	(wersja 1013435)
@@ -113,6 +113,7 @@
     Qt::CaseSensitivity cachedCaseSensitivity;
     bool cachedViewportMove : 1;
     bool cachedNoDialogs : 1;
+    bool isCurrentlySearching : 1;
     QColor cachedColor;
 };
 
@@ -1008,6 +1009,10 @@
     {
         // if the user cancelled but he just got a match, give him the match!
         QApplication::restoreOverrideCursor();
+
+        RunningSearch * s = m_searches[searchID];
+        s->isCurrentlySearching = false;
+
         emit m_parent->searchFinished( searchID, Document::SearchCancelled );
         delete pagesToNotify;
         return;
@@ -1058,11 +1063,13 @@
 
     bool foundAMatch = false;
 
+    RunningSearch * s = m_searches[searchID];
+    s->isCurrentlySearching = false;
+
     // if a match has been found..
     if ( match )
     {
         // update the RunningSearch structure adding this match..
-        RunningSearch * s = m_searches[searchID];
         foundAMatch = true;
         s->continueOnPage = currentPage;
         s->continueOnMatch = *match;
@@ -1110,6 +1117,10 @@
     {
         // if the user cancelled but he just got a match, give him the match!
         QApplication::restoreOverrideCursor();
+
+        RunningSearch * s = m_searches[searchID];
+        s->isCurrentlySearching = false;
+
         emit m_parent->searchFinished( searchID, Document::SearchCancelled );
         delete pagesToNotify;
         return;
@@ -1159,13 +1170,15 @@
     // reset cursor to previous shape
     QApplication::restoreOverrideCursor();
 
+    RunningSearch * s = m_searches[searchID];
+    s->isCurrentlySearching = false;
+
     bool foundAMatch = false;
 
     // if a match has been found..
     if ( match )
     {
         // update the RunningSearch structure adding this match..
-        RunningSearch * s = m_searches[searchID];
         foundAMatch = true;
         s->continueOnPage = currentPage;
         s->continueOnMatch = *match;
@@ -1214,6 +1227,10 @@
         typedef QVector<RegularAreaRect *> MatchesVector;
 
         QApplication::restoreOverrideCursor();
+
+        RunningSearch * s = m_searches[searchID];
+        s->isCurrentlySearching = false;
+
         emit m_parent->searchFinished( searchID, Document::SearchCancelled );
         foreach(const MatchesVector &mv, *pageMatches) qDeleteAll(mv);
         delete pageMatches;
@@ -1256,6 +1273,7 @@
         QApplication::restoreOverrideCursor();
 
         RunningSearch * s = m_searches[searchID];
+        s->isCurrentlySearching = false;
         bool foundAMatch = pageMatches->count() != 0;
         QMap< Page *, QVector<RegularAreaRect *> >::const_iterator it, itEnd;
         it = pageMatches->constBegin();
@@ -1299,6 +1317,10 @@
         typedef QVector<MatchColor> MatchesVector;
 
         QApplication::restoreOverrideCursor();
+
+        RunningSearch * s = m_searches[searchID];
+        s->isCurrentlySearching = false;
+
         emit m_parent->searchFinished( searchID, Document::SearchCancelled );
 
         foreach(const MatchesVector &mv, *pageMatches)
@@ -1372,6 +1394,7 @@
         QApplication::restoreOverrideCursor();
 
         RunningSearch * s = m_searches[searchID];
+        s->isCurrentlySearching = false;
         bool foundAMatch = pageMatches->count() != 0;
         QMap< Page *, QVector<MatchColor> >::const_iterator it, itEnd;
         it = pageMatches->constBegin();
@@ -2547,6 +2570,7 @@
     s->cachedViewportMove = moveViewport;
     s->cachedNoDialogs = noDialogs;
     s->cachedColor = color;
+    s->isCurrentlySearching = true;
 
     // global data for search
     QSet< int > *pagesToNotify = new QSet< int >;
@@ -2645,9 +2669,10 @@
 
     // start search with cached parameters from last search by searchID
     RunningSearch * p = *it;
-    searchText( searchID, p->cachedString, false, p->cachedCaseSensitivity,
-                p->cachedType, p->cachedViewportMove, p->cachedColor,
-                p->cachedNoDialogs );
+    if ( !p->isCurrentlySearching )
+        searchText( searchID, p->cachedString, false, p->cachedCaseSensitivity,
+                    p->cachedType, p->cachedViewportMove, p->cachedColor,
+                    p->cachedNoDialogs );
 }
 
 void Document::continueSearch( int searchID, SearchType type )
@@ -2662,9 +2687,10 @@
 
     // start search with cached parameters from last search by searchID
     RunningSearch * p = *it;
-    searchText( searchID, p->cachedString, false, p->cachedCaseSensitivity,
-                type, p->cachedViewportMove, p->cachedColor,
-                p->cachedNoDialogs );
+    if ( !p->isCurrentlySearching )
+        searchText( searchID, p->cachedString, false, p->cachedCaseSensitivity,
+                    type, p->cachedViewportMove, p->cachedColor,
+                    p->cachedNoDialogs );
 }
 
 void Document::resetSearch( int searchID )
Index: okular/generators/dvi/psgs.cpp
===================================================================
--- okular/generators/dvi/psgs.cpp	(.../tags/KDE/4.3.0/kdegraphics)	(wersja 1013435)
+++ okular/generators/dvi/psgs.cpp	(.../branches/KDE/4.3/kdegraphics)	(wersja 1013435)
@@ -215,6 +215,7 @@
   // Step 2: Call GS with the File
   QFile::remove(filename.toAscii());
   KProcess proc;
+  proc.setOutputChannelMode(KProcess::SeparateChannels);
   QStringList argus;
   argus << "gs";
   argus << "-dSAFER" << "-dPARANOIDSAFER" << "-dDELAYSAFER" << "-dNOPAUSE" << "-dBATCH";
@@ -259,6 +260,7 @@
 	gsDevice = knownDevices.begin();
 	if (knownDevices.isEmpty())
 	  // TODO: show a requestor of some sort.
+#if 0
 	  KMessageBox::detailedError(0, 
 				     i18n("<qt>The version of Ghostview that is installed on this computer does not contain "
 					  "any of the Ghostview device drivers that are known to Okular. PostScript "
@@ -279,6 +281,9 @@
 					  "Ghostview. Among others, Okular can use the 'png256', 'jpeg' and 'pnm' "
 					  "drivers. Note that Okular needs to be restarted to re-enable PostScript support."
 					  "</p></qt>"));
+#else
+	{}
+#endif
 	else {
 	  kDebug(kvs::dvi) << QString("Okular will now try to use the '%1' device driver.").arg(*gsDevice);
 	  gs_generate_graphics_file(page, filename, magnification);
Index: okular/generators/tiff/libokularGenerator_tiff.desktop
===================================================================
--- okular/generators/tiff/libokularGenerator_tiff.desktop	(.../tags/KDE/4.3.0/kdegraphics)	(wersja 1013435)
+++ okular/generators/tiff/libokularGenerator_tiff.desktop	(.../branches/KDE/4.3/kdegraphics)	(wersja 1013435)
@@ -15,6 +15,7 @@
 Name[ga]=Leabharlann TIFF Okular
 Name[gl]=Biblioteca TIFF de Okular
 Name[hne]=ओकुलर टिफ लाइब्रेरी
+Name[hr]=TIFF biblioteka za Okular
 Name[hu]=Okular TIFF-kezelő programkönyvtár
 Name[is]=Okular TIFF skráasafn
 Name[it]=Libreria TIFF di Okular
Index: okular/generators/comicbook/libokularGenerator_comicbook.desktop
===================================================================
--- okular/generators/comicbook/libokularGenerator_comicbook.desktop	(.../tags/KDE/4.3.0/kdegraphics)	(wersja 1013435)
+++ okular/generators/comicbook/libokularGenerator_comicbook.desktop	(.../branches/KDE/4.3/kdegraphics)	(wersja 1013435)
@@ -13,6 +13,7 @@
 Name[gl]=Libro de Cómic
 Name[hi]=कॉमिक पुस्तिका
 Name[hne]=कामिक पुस्तिका
+Name[hr]=Strip
 Name[hu]=Képregénynézegető
 Name[km]=សៀវភៅ​កំប្លែង
 Name[ko]=만화책
Index: kolourpaint/document/kpDocument_Open.cpp
===================================================================
--- kolourpaint/document/kpDocument_Open.cpp	(.../tags/KDE/4.3.0/kdegraphics)	(wersja 1013435)
+++ kolourpaint/document/kpDocument_Open.cpp	(.../branches/KDE/4.3/kdegraphics)	(wersja 1013435)
@@ -217,6 +217,13 @@
             detectedMimeTypePtr != KMimeType::defaultMimeTypePtr ())
         {
             detectedMimeType = detectedMimeTypePtr->name ();
+        } else {
+            detectedMimeTypePtr = KMimeType::findByPath (tempFile);
+            if(detectedMimeTypePtr &&
+                detectedMimeTypePtr != KMimeType::defaultMimeTypePtr ())
+            {
+                detectedMimeType = detectedMimeTypePtr->name ();
+            }
         }
 
         if (saveOptions)
Index: ksnapshot/regiongrabber.cpp
===================================================================
--- ksnapshot/regiongrabber.cpp	(.../tags/KDE/4.3.0/kdegraphics)	(wersja 1013435)
+++ ksnapshot/regiongrabber.cpp	(.../branches/KDE/4.3/kdegraphics)	(wersja 1013435)
@@ -28,7 +28,8 @@
 #include <KWindowSystem>
 
 RegionGrabber::RegionGrabber( ) :
-    QWidget( 0 ), selection(), mouseDown( false ), newSelection( false ),
+    QWidget( 0, Qt::X11BypassWindowManagerHint | Qt::WindowStaysOnTopHint | Qt::FramelessWindowHint | Qt::Tool ),
+    selection(), mouseDown( false ), newSelection( false ),
     handleSize( 10 ), mouseOverHandle( 0 ),
     showHelp( true ), grabbing( false ),
     TLHandle(0,0,handleSize,handleSize), TRHandle(0,0,handleSize,handleSize),
@@ -39,7 +40,6 @@
     handles << &TLHandle << &TRHandle << &BLHandle << &BRHandle
             << &LHandle << &THandle << &RHandle << &BHandle;
     setMouseTracking( true );
-    setWindowFlags( Qt::WindowStaysOnTopHint | Qt::FramelessWindowHint );
     int timeout = KWindowSystem::compositingActive() ? 200 : 50;
     QTimer::singleShot( timeout, this, SLOT(init()) );
 }
@@ -54,8 +54,9 @@
     resize( pixmap.size() );
     move( 0, 0 );
     setCursor( Qt::CrossCursor );
-    showFullScreen();
-    KWindowSystem::forceActiveWindow( winId() );
+    show();
+    grabMouse();
+    grabKeyboard();
 }
 
 void RegionGrabber::paintEvent( QPaintEvent* e )

Zmiany atrybutów dla: .
___________________________________________________________________
Dodane: svn:externals
   + 


